build/common.go

```diff
@@ -76,24 +76,35 @@ func buildBackend(cfg Config) error {
 		return err
 	}
 
-	// Link statically
-	ldFlags := `-extldflags "-static"`
-	if !cfg.EnableDebug {
-		// Add linker flags to drop debug information
-		ldFlags = "-w -s " + ldFlags
-	}
-
 	args := []string{
-		"build", "-o", path.Join("dist", exeName), "-ldflags", ldFlags,
+		"build", "-o", path.Join("dist", exeName),
+	}
+	if !cfg.EnableDynamicLinking {
+		// Link statically
+		ldFlags := `-extldflags "-static"`
+		if !cfg.EnableDebug {
+			// Add linker flags to drop debug information
+			ldFlags = "-w -s " + ldFlags
+		}
+		args = append(args, "-ldflags", ldFlags)
+	} else {
+		args = append(args, "-tags", "netgo")
+		if !cfg.EnableDebug {
+			args = append(args, "-ldflags", "-w")
```

Comment:
Why do the ld flags here differ from the static linking case?

Reply:
I just tried to imitate the previous behaviour [here](https://github.com/grafana/grafana-plugin-sdk-go/commit/de2048a3ee264b800e542a956ee9bd9362f36967#diff-a2e1f6ccdeb2552b19bf114519adcea9L116) (before static linking was enforced)

Reply:
@papagian I think we should apply the `-w` and `-s` flags independently of whether we use cgo/link statically or not, since the the former turns off DWARF debugging info and the latter removes the Go symbol table, both to reduce binary size.